staging/src/k8s.io/cri-client/pkg/logs/logs_test.go

```diff
@@ -596,3 +596,61 @@ func TestOnlyStdoutStream(t *testing.T) {
 		})
 	}
 }
+
+func TestDedupEvents(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    []interface{}
+		expected []interface{}
+	}{
+		{
+			name: "events without error",
+			input: []interface{}{fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Create,
+				fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Create,
+				fsnotify.Write, fsnotify.Write, fsnotify.Create},
+			expected: []interface{}{fsnotify.Write, fsnotify.Write, fsnotify.Create,
+				fsnotify.Write, fsnotify.Create,
+				fsnotify.Write, fsnotify.Create},
+		},
+		{
+			name:     "event with error",
+			input:    []interface{}{fsnotify.Write, fsnotify.Write, fsnotify.Write, fsnotify.Create, fsnotify.ErrEventOverflow, fsnotify.ErrEventOverflow},
+			expected: []interface{}{fsnotify.Write, fsnotify.Write, fsnotify.Create},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			w := &fsnotify.Watcher{Events: make(chan fsnotify.Event), Errors: make(chan error)}
+			dedup := newDedupWriteEvents(w, "0.log", nil)
+			dedup.dedupEvents(context.Background())
```

Comment:
I'd suggest to use ktesting.Init(t) to get a text context in the beginning of the test and then reuse it in the loop.

Reply:
Sorry,  I can not understand what you mean.

Reply:
Sorry for being unclear. I suggested to have `tCtx := ktesting.Init(t)` in the beginning of the test and then use it here instead of `context.Background()` this way: `dedup.dedupEvents(tCtx)`. This is a common approach in k/k unit tests, for example https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_test.go#L3361

Reply:
I've removed the `ctx` parameter. It's no longer necessary.